skimage/segmentation/random_walker_segmentation.py

```diff
@@ -146,6 +148,7 @@ def _build_laplacian(data, spacing, mask, beta, multichannel):
     j_indices = edges[::-1].ravel()
     data = np.hstack((weights, weights))
     lap = sparse.csr_array((data, (i_indices, j_indices)), shape=(pixel_nb, pixel_nb))
+    lap = sparse.lil_array(lap)  # Avoid SparseEfficiencyWarning
```

Comment:
For SciPy  1.11.2, the line below (`.setdiag`) previously raised

> SparseEfficiencyWarning: Changing the sparsity structure of a csr_matrix is expensive. lil_matrix is more efficient.

Could somebody with more background in scipy.sparse check that this is the right approach... cc @dschult, @stefanv

Reply:
We chose to revert this fix and ignore the warning like previously since the warning seems to go away on later SciPy versions.